### PR TITLE
Meta: convert webspeechapi.html to Bikeshed

### LINKED ISSUE #3: Convert to Bikeshed
https://github.com/tabatkins/bikeshed is used for most modern specs.

### LINKED ISSUE #4: Implement gh-pages branch

### LINKED ISSUE #5: Implement travis-ci to run tests and deploy automatically

### PR DESCRIPTION
**Pull request by foolip**
https://github.com/foolip/epubbase/blob/master/worddiff.py was used to
vet that the text content of the spec wasn't changed in any unexpected
way.

The only notable change is that "message attribute" in
https://w3c.github.io/speech-api/webspeechapi.html#speechsynthesiserrorevent
was removed, as it was a copy-paste error and SpeechSynthesisErrorEvent
doesn't have a message attribute, just an error attribute.

Fixes #3. Fixes #4. Fixes #5.

### CODE REVIEW

**Comment by foolip** [comment]
@andrenatal @saschanaz please review.

To make it easier to review this, the gh-pages branch was created by https://travis-ci.org/w3c/speech-api/builds/391819131 before I restricted it to master. https://w3c.github.io/speech-api/ therefore already contains the output of this, but that isn't the version that is usually linked to.

While I've compared the text content extensively, I haven't paid a lot of attention to visual changes, so please review with care.

File: index.bs [hunk 1]
```diff
@@ -7,8 +7,10 @@ Group: SACG
 URL: https://w3c.github.io/speech-api/
 Repository: w3c/speech-api
 Shortname: speech-api
-Editor: Glen Shires (Google)
-Editor: Hans Wennborg (Google)
+Editor: Andre Natal (Mozilla)
+Editor: Philip Jägenstedt (Google)
+Former Editor: Glen Shires (Google)
```
Comment (foolip) [inline index.bs]: @gshires if you do still intend to do any work on this spec please let me know and I'll revert the "Former" bit here. This seems to reflect the current state of things, but I'm not trying to make more work for myself if not necessary :)

Comment (foolip) [reply index.bs]: Searching for something else in my inbox I came across the thread about setting up this repo where you said "I'd like to remain as an editor" so I'll revert this bit as well. I can't find Hans on GitHub at all so will leave him as former.

**Comment by saschanaz** [comment]
Can IDL index be enabled?

We may ultimately split 4.1 and 4.2 into smaller pieces as they includes several different interfaces, but maybe out of scope for this PR.

**Comment by saschanaz** [comment]
image

It seems `MUST`s are not properly rendered as `small-caps`.

**Comment by foolip** [comment]
> Can IDL index be enabled?

I've enabled all indexes, see https://w3c.github.io/speech-api/

> ~~It seems `MUST`s are not properly rendered as `small-caps`.~~ [Other W3C specs](https://w3c.github.io/ServiceWorker/) do the same thing, maybe is just okay.

I would propose that we remove all markup around these keywords entirely. Many modern specs do it this way, see https://drafts.csswg.org/web-animations/ for an example. Bikeshed also warns if you use these keywords in examples or non-normative places. Note that currently not all "must" are annotated. Will push a commit to simplify.

**Comment by saschanaz** [comment]
image

IDL Index has no bottom margin, seems problematic.

**Comment by foolip** [comment]
> IDL Index has no bottom margin, seems problematic

Can you file a bug at https://github.com/tabatkins/bikeshed/issues/new about that? I haven't tweaked the styling, so it's the default Bikeshed look.

**Comment by foolip** [comment]
Filed https://github.com/validator/validator/issues/667 about remaining HTML checker issues.

File: index.bs [hunk 2]
```diff
@@ -7,7 +7,6 @@ Group: SACG
 URL: https://w3c.github.io/speech-api/
 Repository: w3c/speech-api
 Shortname: speech-api
-Editor: Andre Natal (Mozilla)
```
Comment (foolip) [inline index.bs]: @andrenatal didn't want to add you without confirmation on the PR, please add yourself back with the first spec change that you make after this is merged.

**Comment by foolip** [comment]
I'm about to merge this now, showing the output of worddiff.py mentioned in the commit message for other to verify. I dropped the final part of the diff which is the addition of large indexes.

```diff
--- webspeechapi.html
+++ index.html
@@ -1,62 +1,56 @@
 Web Speech API
-Specification Editor's Draft: 11
+Draft Community Group Report, 15
 June 2018
+This version: https://w3c.github.io/speech-api/ Issue Tracking: GitHub
 Editors: Glen
-Shires, Google Inc.
+Shires (Google) Philip Jägenstedt (Google) Former Editor:
 Hans
-Wennborg, Google Inc. This document contains the 19 October 2012 Web Speech API Specification with its errata (E01-E16) applied.
+Wennborg (Google)
 Copyright ©
-2014
+2018
 the Contributors to the Web Speech API Specification, published by the Speech API Community Group under the W3C Community Contributor License Agreement (CLA). A human-readable
@@ -105,52 +99,52 @@
 for forms, continuous dictation and control. The JavaScript API allows web pages to control activation and timing and to handle results and alternatives. Status of
-This Document
+this document
 This specification was published by the Speech API Community Group. It is not a W3C Standard nor is it on the W3C Standards Track. Please
@@ -170,356 +164,150 @@
 limited opt-out and other conditions apply. Learn more about W3C Community and Business Groups. All feedback is welcome: public-speech-api@w3.org (Public archive). Table of Contents 1
-Conformance requirements
+Introduction
 2
-Introduction 3
 Use Cases
-4
+3
 Security and privacy considerations
-5
+3.1 Implementation considerations 4
 API Description
-5.1
+4.1
 The SpeechRecognition Interface
-5.1.1
+4.1.1
 SpeechRecognition Attributes
-5.1.2
+4.1.2
 SpeechRecognition Methods
-5.1.3
+4.1.3
 SpeechRecognition Events
-5.1.4
+4.1.4
 SpeechRecognitionError
-5.1.5
+4.1.5
 SpeechRecognitionAlternative
-5.1.6
+4.1.6
 SpeechRecognitionResult
-5.1.7
+4.1.7
 SpeechRecognitionResultList
-5.1.8
+4.1.8
 SpeechRecognitionEvent
-5.1.9
+4.1.9
 SpeechGrammar
-5.1.10
+4.1.10
 SpeechGrammarList
-5.2
+4.2
 The SpeechSynthesis Interface
-5.2.1
+4.2.1
 SpeechSynthesis Attributes
-5.2.2
+4.2.2
 SpeechSynthesis Methods
-5.2.2.1
+4.2.3
 SpeechSynthesis Events
-5.2.3
+4.2.4
 SpeechSynthesisUtterance Attributes
-5.2.4
+4.2.5
 SpeechSynthesisUtterance Events
-5.2.5
+4.2.6
 SpeechSynthesisEvent Attributes
-5.2.5.1
+4.2.7
 SpeechSynthesisErrorEvent Attributes
-5.2.6
+4.2.8
 SpeechSynthesisVoice
-6
+Attributes 5
 Examples
-6.1
+5.1
 Speech Recognition Examples
-6.2
+5.2
 Speech Synthesis Examples Acknowledgments
-References 1
 Conformance
-requirements All diagrams, examples, and notes in
+Index Terms defined by
 this specification
-are non-normative, as are all sections explicitly marked non-normative. Everything else in this specification is normative. The key words "MUST", "MUST NOT", "REQUIRED", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in the normative parts of this document are to be interpreted as described in RFC2119. For readability, these words do not appear in all uppercase letters in this specification. [RFC2119] Requirements phrased in the imperative as part of algorithms (such as "strip any leading space characters" or "return false and abort these steps") are to be interpreted with the meaning of the key word ("must", "should", "may", etc) used in introducing the algorithm. Conformance requirements phrased as algorithms or specific steps may be implemented in any manner, so long as the end result is equivalent. (In particular, the algorithms
+Terms
 defined
-in this specification are intended to be easy to follow, and not intended to be performant.) User agents may impose implementation-specific limits on otherwise unconstrained inputs, e.g. to prevent denial of service attacks, to guard against running out of memory, or to work around platform-specific limitations. Implementations that use ECMAScript to implement the APIs defined in this specification must implement them in a manner consistent with the ECMAScript Bindings defined in the Web
+by reference References Normative References Informative References
 IDL
-specification, as this specification uses that specification's terminology. [WEBIDL] 2
+Index 1.
 Introduction This section is non-normative. The Web Speech API aims to enable web developers to provide, in a web browser, speech-input and text-to-speech output features
@@ -705,51 +493,51 @@
 future standardization of additions to the markup, API or underlying transport protocols, and indeed the Incubator Report defines a potential roadmap for such future work.
-3
+2.
 Use Cases This section is non-normative. This specification supports the following use cases, as defined in Section 4 of the Incubator Report. Voice Web Search
@@ -872,51 +660,51 @@
 avoid the need for Rerecognition by using a larger grammar, or by combining multiple grammars — both of these techniques are supported in this specification.
-4
+3.
 Security and privacy considerations User agents must only start speech input sessions with explicit, informed user consent. User consent can include, for example: User click
@@ -1098,50 +886,51 @@
 input is used, to let the user now what it is and how they can tune their privacy settings to disable speech recording if required.
+3.1.
 Implementation considerations This section is non-normative. Spoken password inputs can be problematic from a security perspective, but it is up to the user to decide
@@ -1213,104 +1002,103 @@
 An example of styling the file input element can be seen at http://www.quirksmode.org/dom/inputfile.html. The above recommendations are intended to reduce this risk of such attacks.
-5
+4.
 API Description This section is normative.
-5.1
+4.1.
 The SpeechRecognition Interface The speech recognition interface is the scripted web API for controlling a given recognition. The term "final result" indicates a SpeechRecognitionResult in which the final attribute is true. The term "interim result" indicates a SpeechRecognitionResult in which the final attribute is false.
-IDL
 [Exposed=Window, Constructor] interface SpeechRecognition : EventTarget { // recognition parameters attribute SpeechGrammarList grammars; attribute DOMString lang; attribute boolean continuous; attribute boolean interimResults; attribute unsigned long
@@ -1536,141 +1324,150 @@
 SpeechGrammarList { readonly attribute unsigned long length; getter SpeechGrammar item(unsigned long index); void addFromURI(DOMString src, optional float weight); void addFromString(DOMString string, optional float weight); };
-5.1.1
+4.1.1.
 SpeechRecognition Attributes grammars
-attribute
+attribute, of type SpeechGrammarList
 The grammars attribute stores the collection of SpeechGrammar objects which represent the grammars that are active for this recognition. lang
-attribute
+attribute, of type DOMString
 This attribute will set the language of the recognition for the request, using a valid BCP 47 language tag. [BCP47] If unset it remains unset for getting in script, but will default to use the
-lang
+language
 of the html document root element and associated hierarchy. This default value is computed and used when the input request opens a connection to the recognition service. continuous
-attribute
+attribute, of type boolean
 When the continuous attribute is set to false, the user agent must return no more than one final result in response to starting recognition, for
@@ -1706,111 +1503,121 @@
 in response to starting recognition, for example a dictation. The default value must be false. Note, this attribute setting does not affect interim results. interimResults
-attribute
+attribute, of type boolean
 Controls whether interim results are returned. When set to true, interim results should be returned. When set to false, interim results must
-NOT
+not
 be returned. The default value must be false. Note, this attribute setting does not affect final results. maxAlternatives
-attribute
+attribute, of type unsigned long
 This attribute will set the maximum number of SpeechRecognitionAlternatives per result. The default value is 1. serviceURI
-attribute
+attribute, of type DOMString
 The serviceURI attribute specifies the location of the speech recognition service that the web application wishes to use. If this attribute is unset at the
@@ -1901,54 +1708,54 @@
 browser configuration parameters. <i>[Editor note: The group is currently discussing whether WebRTC might be used to specify selection of audio sources and remote recognizers.] [5]</i>
-5.1.2
+4.1.2.
 SpeechRecognition Methods
-start
+start()
 method When the start method is called it represents the moment in time the web application wishes to begin recognition. When the speech input is
@@ -2015,57 +1822,57 @@
 object (that is, start has previously been called, and no error or end event has fired on the object), the user agent must throw an
-InvalidStateError exception
+"InvalidStateError" DOMException
 and ignore the call.
-stop
+stop()
 method The stop method represents an instruction to the recognition service to stop listening to more audio, and to try and return a result using
@@ -2230,51 +2037,51 @@
 on it, the end or error event has fired on it, or stop was previously called on it), the user agent must ignore the call.
-abort
+abort()
 method The abort method is a request to immediately stop listening and stop recognizing and do not return any information but that the system is
@@ -2331,51 +2138,51 @@
 on it, the end or error event has fired on it, or abort was previously called on it), the user agent must ignore the call.
-5.1.3
+4.1.3.
 SpeechRecognition Events The DOM Level 2 Event Model is used for speech recognition events. The methods in the EventTarget interface should be used for registering
@@ -2427,51 +2234,51 @@
 set to the best possible estimate of when the real-world event which the event object represents occurred. This timestamp must be represented in the user
-agent's
+agent’s
 view of time, even for events where the timestamps in question could be raised on a different machine like a remote recognition service (i.e., in
@@ -2802,65 +2609,69 @@
 end event Fired when the service has disconnected. The event must always be generated when the session ends no matter the reason for the end.
-5.1.4
+4.1.4.
 SpeechRecognitionError The SpeechRecognitionError event is the interface used for the error event. error
-attribute
+attribute, of type SpeechRecognitionErrorCode, readonly
 The errorCode is an enumeration indicating what has gone wrong. The values are: "no-speech" No speech was detected. "aborted" Speech input was aborted somehow, maybe
@@ -2919,179 +2730,191 @@
 user agent is not allowing the web application requested speech service, but would allow some speech service, to be used either because the user agent
-doesn't
+doesn’t
 support the selected one or because of reasons of security, privacy or user preference. "bad-grammar" There was an error in the speech recognition grammar or semantic tags, or the grammar format or semantic tag format is unsupported. "language-not-supported" The language was not supported. message
-attribute
+attribute, of type DOMString, readonly
 The message content is implementation specific. This attribute is primarily intended for debugging and developers should not use it directly in their application user interface.
-5.1.5
+4.1.5.
 SpeechRecognitionAlternative The SpeechRecognitionAlternative represents a simple view of the response that gets used in a n-best list. transcript
-attribute
+attribute, of type DOMString, readonly
 The transcript string represents the raw words that the user spoke. For continuous recognition, leading or trailing whitespace MUST be included where necessary such that concatenation of consecutive SpeechRecognitionResults produces a proper transcript of the session. confidence
-attribute
+attribute, of type float, readonly
 The confidence represents a numeric estimate between 0 and 1 of how confident the recognition system is that the recognition is correct. A higher number
@@ -3112,97 +2935,102 @@
 confidence can be specified in a speech-recognition-engine-independent manner and whether confidence threshold and nomatch should be included, because this is not a dialog API.] [4]</i>
-5.1.6
+4.1.6.
 SpeechRecognitionResult The SpeechRecognitionResult object represents a single one-shot recognition match, either as one small part of a continuous recognition or as the complete return result of a non-continuous recognition. length
-attribute
+attribute, of type unsigned long, readonly
 The long attribute represents how many n-best alternatives are represented in the item array.
-item
+item(index)
 getter The item getter returns a SpeechRecognitionAlternative from the index into an array of n-best values. If index is greater than or equal to length,
@@ -3237,225 +3065,243 @@
 the n-best list is sorted in non-increasing confidence order (each element must be less than or equal to the confidence of the preceding elements). isFinal
-attribute
+attribute, of type boolean, readonly
 The final boolean must be set to true if this is the final time the speech service will return this particular index value. If the value is false, then this represents an interim result that could still be changed.
-5.1.7
+4.1.7.
 SpeechRecognitionResultList The SpeechRecognitionResultList object holds a sequence of recognition results representing the complete return result of a continuous recognition. For a non-continuous recognition it will hold only a single value. length
-attribute
+attribute, of type unsigned long, readonly
 The length attribute indicates how many results are represented in the item array.
-item
+item(index)
 getter The item getter returns a SpeechRecognitionResult from the index into an array of result values. If index is greater than or equal to length, this returns null. The user agent must ensure that the length attribute is set to the number of elements in the array.
-5.1.8
+4.1.8.
 SpeechRecognitionEvent The SpeechRecognitionEvent is the event that is raised each time there are any changes to interim or final results. resultIndex
-attribute
+attribute, of type unsigned long, readonly
 The resultIndex must be set to the lowest index in the "results" array that has changed. results
-attribute
+attribute, of type SpeechRecognitionResultList, readonly
 The array of all current recognition results for this session. Specifically all final results that have been returned, followed by the current best hypothesis for
@@ -3597,51 +3443,55 @@
 when resultIndex equals results.length, no new results are returned, this may occur when the array length decreases to remove one or more interim results. interpretation
-attribute
+attribute, of type any, readonly
 The interpretation represents the semantic meaning from what the user said. This might be determined, for instance, through the SISR specification of semantics in a
@@ -3651,60 +3501,64 @@
 The group is currently discussing options for the value of the interpretation attribute when no interpretation has been returned by the recognizer. Current options are
-'null'
+null
 or a copy of the transcript.] [2]</i> emma
-attribute
+attribute, of type Document, readonly
 EMMA 1.0 representation of this result. [EMMA] The contents of this result could vary across user agents and recognition engines, but all implementations must expose
@@ -3753,201 +3607,212 @@
 utilized for recognition, except where infeasible due to conflicting attributes. The user agent may add additional annotations to provide a richer result for the developer.
-5.1.9
+4.1.9.
 SpeechGrammar The SpeechGrammar object represents a container for a grammar. <i>[Editor note: The group is currently discussing options for which grammar formats should be supported, how builtin grammar types are specified, and default grammars when not specified.] [2] [3]</i> This structure has the following attributes: src
-attribute
+attribute, of type DOMString
 The required src attribute is the URI for the grammar. Note some services may support builtin grammars that can be specified using a builtin URI scheme. weight
-attribute
+attribute, of type float
 The optional weight attribute controls the weight that the speech recognition service should use with this grammar. By default, a grammar has a weight of 1. Larger weight values positively weight the grammar while smaller weight values make the grammar weighted less strongly.
-5.1.10
+4.1.10.
 SpeechGrammarList The SpeechGrammarList object represents a collection of SpeechGrammar objects. This structure has the following attributes: length
-attribute
+attribute, of type unsigned long, readonly
 The length attribute represents how many grammars are currently in the array.
-item
+item(index)
 getter The item getter returns a SpeechGrammar from the index into an array of grammars. The user agent must ensure that the length attribute is
@@ -3962,51 +3827,52 @@
 array. The user agent must ensure that the index order from smallest to largest matches the order in which grammars were added to the array.
-addFromURI
+addFromURI(src, weight)
 method This method appends a grammar to the grammars array parameter based on URI. The URI for the grammar is specified by the src parameter,
@@ -4017,160 +3883,160 @@
 for the grammar. Note, some services may support builtin grammars that can be specified by URI. If the weight parameter is present it represents this
-grammar's
+grammar’s
 weight relative to the other grammar. If the weight parameter is not present, the default value of 1.0 is used.
-addFromString
+addFromString(string, weight)
 method This method appends a grammar to the grammars array parameter based on text. The content of the grammar is specified by the string parameter. This content should be encoded into a data: URI when the SpeechGrammar object is created. If the weight parameter is present it represents this
-grammar's
+grammar’s
 weight relative to the other grammar. If the weight parameter is not present, the default value of 1.0 is used.
-5.2
+4.2.
 The SpeechSynthesis Interface The SpeechSynthesis interface is the scripted web API for controlling a text-to-speech output.
-IDL
 [Exposed=Window] interface SpeechSynthesis : EventTarget { readonly attribute boolean pending; readonly attribute boolean speaking; readonly attribute boolean paused; attribute EventHandler onvoiceschanged; void speak(SpeechSynthesisUtterance utterance); void
@@ -4294,170 +4160,183 @@
 [Exposed=Window] interface SpeechSynthesisVoice { readonly attribute DOMString voiceURI; readonly attribute DOMString name; readonly attribute DOMString lang; readonly attribute boolean localService; readonly attribute boolean default; };
-5.2.1
+4.2.1.
 SpeechSynthesis Attributes pending
-attribute
+attribute, of type boolean, readonly
 This attribute is true if the queue for the global SpeechSynthesis instance contains any utterances which have not started speaking. speaking
-attribute
+attribute, of type boolean, readonly
 This attribute is true if an utterance is being spoken. Specifically if an utterance has begun being spoken and has not completed being spoken. This is independent of whether the global SpeechSynthesis instance is in the paused state. paused
-attribute
+attribute, of type boolean, readonly
 This attribute is true when the global SpeechSynthesis instance is in the paused state. This state is independent of whether anything is in the queue. The default state of a the global SpeechSynthesis instance for a new window is the non-paused state.
-5.2.2
+4.2.2.
 SpeechSynthesis Methods
-speak
+speak(utterance)
 method This method appends the SpeechSynthesisUtterance object
+utterance
 to the end of the queue for the global SpeechSynthesis instance. It does not change the paused state of the SpeechSynthesis instance. If the SpeechSynthesis
@@ -4565,120 +4444,120 @@
 argument to another SpeechSynthesis object should throw an exception. (For example, two frames may have the same origin and each will contain a SpeechSynthesis object.)
-cancel
+cancel()
 method This method removes all utterances from the queue. If an utterance is being spoken, speaking ceases immediately. This method does not change the paused state of the global SpeechSynthesis instance.
-pause
+pause()
 method This method puts the global SpeechSynthesis instance into the paused state. If an utterance was being spoken, it pauses mid-utterance. (If called when the SpeechSynthesis instance was already in the paused state, it does nothing.)
-resume
+resume()
 method This method puts the global SpeechSynthesis instance into the non-paused state. If an utterance was speaking, it continues speaking the utterance at the point
@@ -4692,51 +4571,51 @@
 begins speaking the next utterance in the queue (if any). (If called when the SpeechSynthesis instance was already in the non-paused state, it does nothing.)
-getVoices
+getVoices()
 method This method returns the available voices. It is user agent dependent which voices are available. If there are no voices available, or if the
@@ -4747,91 +4626,94 @@
 voices is not yet known (for example: server-side synthesis where the list is determined asynchronously), then this method must return a SpeechSynthesisVoiceList of length zero.
-5.2.2.1
+4.2.3.
 SpeechSynthesis Events voiceschanged event Fired when the contents of the SpeechSynthesisVoiceList, that the getVoices method will return, have changed. Examples include: server-side synthesis where the list is determined asynchronously, or when client-side voices are installed/uninstalled.
-5.2.3
+4.2.4.
 SpeechSynthesisUtterance Attributes text
-attribute
+attribute, of type DOMString
 This attribute specifies the text to be synthesized and spoken for this utterance. This may be either plain text or a complete, well-formed SSML document.
@@ -4864,116 +4746,122 @@
 they do not support and speak the text. There may be a maximum length of the text, it may be limited to 32,767 characters. lang
-attribute
+attribute, of type DOMString
 This attribute specifies the language of the speech synthesis for the utterance, using a valid BCP 47 language tag. [BCP47] If unset it remains unset for getting in script, but will default to use the
-lang
+language
 of the html document root element and associated hierarchy. This default value is computed and used when the input request opens a connection to the recognition service. voice
-attribute
+attribute, of type SpeechSynthesisVoice
 The voice attribute specifies speech synthesis voice that the web application wishes to use. If, at the time of the play method call, this attribute
@@ -5042,100 +4930,106 @@
 a local or remote speech service and can incorporate end user choices via interfaces provided by the user agent such as browser configuration parameters. volume
-attribute
+attribute, of type float
 This attribute specifies the speaking volume for the utterance. It ranges between 0 and 1 inclusive, with 0 being the lowest volume and 1 the highest volume, with a default of 1. If SSML is used, this value will be overridden by prosody tags in the markup. rate
-attribute
+attribute, of type float
 This attribute specifies the speaking rate for the utterance. It is relative to the default rate for this voice. 1 is the default rate supported
@@ -5207,51 +5101,54 @@
 even if you specify a value larger than 3. If SSML is used, this value will be overridden by prosody tags in the markup. pitch
-attribute
+attribute, of type float
 This attribute specifies the speaking pitch for the utterance. It ranges between 0 and 2 inclusive, with 0 being the lowest pitch and 2 the
@@ -5277,51 +5174,51 @@
 or voices may constrain the minimum and maximum rates further. If SSML is used, this value will be overridden by prosody tags in the markup.
-5.2.4
+4.2.5.
 SpeechSynthesisUtterance Events Each of these events must use the SpeechSynthesisEvent interface, except the error event which must use the SpeechSynthesisErrorEvent interface. These events bubble up
@@ -5429,51 +5326,51 @@
 instance is in the paused state, and then calling the resume method does not cause the resume event to be fired, in this case the
-utterance's
+utterance’s
 start event will be called when the utterance starts. mark event Fired when the spoken utterance reaches a named "mark" tag in SSML. [SSML] The
@@ -5497,66 +5394,75 @@
 when the spoken utterance reaches a word or sentence boundary. The user agent must fire this event if the speech synthesis engine provides the event.
-5.2.5
+4.2.6.
 SpeechSynthesisEvent Attributes utterance
-attribute
+attribute, of type SpeechSynthesisUtterance, readonly
 This attribute contains the SpeechSynthesisUtterance that triggered this event. charIndex
-attribute
+attribute, of type unsigned long, readonly
 This attribute indicates the zero-based character index into the original utterance string that most closely approximates the current speaking position of the speech engine. No
@@ -5610,51 +5516,55 @@
 yet been spoken. The user agent must return this value if the speech synthesis engine supports it, otherwise the user agent must return undefined. elapsedTime
-attribute
+attribute, of type float, readonly
 This attribute indicates the time, in seconds, that this event triggered, relative to when this utterance has begun to be spoken. The user agent must
@@ -5662,117 +5572,125 @@
 this value if the speech synthesis engine supports it or the user agent can otherwise determine it, otherwise the user agent must return undefined. name
-attribute
+attribute, of type DOMString, readonly
 For mark events, this attribute indicates the name of the marker, as defined in SSML as the name attribute of a mark element. [SSML] For boundary events, this attribute indicates the type of boundary that caused the event: "word" or "sentence". For all other events, this value should return undefined.
-5.2.5.1
+4.2.7.
 SpeechSynthesisErrorEvent Attributes The SpeechSynthesisErrorEvent is the interface used for the SpeechSynthesisUtterance error event. error
-attribute
+attribute, of type SpeechSynthesisErrorCode, readonly
 The errorCode is an enumeration indicating what has gone wrong. The values are: "canceled" A cancel method call caused the SpeechSynthesisUtterance to be removed from
@@ -5952,81 +5870,59 @@
 the SpeechSynthesisUtterance text attribute is too long to synthesize. "invalid-argument" The contents of the SpeechSynthesisUtterance rate, pitch or volume attribute is not supported by synthesizer.
-message attribute The message content is implementation specific. This attribute is primarily intended for debugging and developers should not use it directly in their application user interface. 5.2.6
+4.2.8.
 SpeechSynthesisVoice
+Attributes
 voiceURI
-attribute
+attribute, of type DOMString, readonly
 The voiceURI attribute specifies the speech synthesis voice and the location of the speech synthesis service for this voice. Note that the voiceURI is a
@@ -6046,90 +5942,102 @@
 use of a URN with meaning to the user agent or by specifying a URL that the user agent recognizes as a local service. name
-attribute
+attribute, of type DOMString, readonly
 This attribute is a human-readable name that represents the voice. There is no guarantee that all names returned are unique. lang
-attribute
+attribute, of type DOMString, readonly
 This attribute is a BCP 47 language tag indicating the language of the voice. [BCP47] localService
-attribute
+attribute, of type boolean, readonly
 This attribute is true for voices supplied by a local speech synthesizer, and is false for voices supplied by a remote speech synthesizer service. (This
@@ -6144,156 +6052,156 @@
 additional latency, bandwidth or cost, whereas local voices may imply lower quality, however there is no guarantee that any of these implications are true.) default
-attribute
+attribute, of type boolean, readonly
 This attribute is true for at most one voice per language. There may be a different default for each language. It is user agent dependent how default voices are determined.
-6
+5.
 Examples This section is non-normative.
-6.1
+5.1.
 Speech Recognition Examples Using speech recognition to fill an input-field and perform a web search.
-Example 1
 <script type="text/javascript"> var recognition = new SpeechRecognition(); recognition.onresult = function(event) { if (event.results.length > 0) { q.value = event.results[0][0].transcript; q.form.submit(); } } </script> <form action="http://www.example.com/search"> <input type="search" id="q" name="q" size=60> <input type="button" value="Click to Speak" onclick="recognition.start()"> </form> Using speech recognition to fill an options list with alternative speech results.
-Example 2
 <script type="text/javascript"> var recognition = new SpeechRecognition(); recognition.maxAlternatives = 10; recognition.onresult = function(event) { if (event.results.length > 0) { var result = event.results[0]; for (var
@@ -6318,52 +6226,50 @@
 } } function start() { select.options.length = 0; recognition.start(); } </script> <select id="select"></select> <button onclick="start()">Click to Speak</button> Using continuous speech recognition to fill a textarea.
-Example 3
 <textarea id="textarea" rows=10 cols=80></textarea> <button id="button" onclick="toggleStartStop()"></button> <script type="text/javascript"> var recognizing; var recognition = new SpeechRecognition(); recognition.continuous = true; reset(); recognition.onend = reset; recognition.onresult =
@@ -6416,52 +6322,50 @@
 recognizing = true; button.innerHTML = "Click to Stop"; } } </script> Using continuous speech recognition, showing final results in black and interim results in grey.
-Example 4
 <button id="button" onclick="toggleStartStop()"></button> <div style="border:dotted;padding:10px"> <span id="final_span"></span> <span id="interim_span" style="color:grey"></span> </div> <script type="text/javascript"> var recognizing; var recognition = new SpeechRecognition(); recognition.continuous = true; recognition.interimResults =
@@ -6533,72 +6437,68 @@
 (recognizing) { recognition.stop(); reset(); } else { recognition.start(); recognizing = true; button.innerHTML = "Click to Stop"; final_span.innerHTML = ""; interim_span.innerHTML = ""; } } </script>
-6.2
+5.2.
 Speech Synthesis Examples Spoken text.
-Example 1
 <script type="text/javascript"> speechSynthesis.speak(new SpeechSynthesisUtterance('Hello World')); </script> Spoken text with attributes and events.
-Example 2
 <script type="text/javascript"> var u = new SpeechSynthesisUtterance(); u.text = 'Hello World'; u.lang = 'en-US'; u.rate = 1.2; u.onend = function(event) { alert('Finished in ' +
@@ -6651,225 +6551,2791 @@
 Inc. Kagami Sascha Rosylight Also, the members of the HTML Speech Incubator Group, and the corresponding Final Report [1], created the basis for this specification.
```

**Pull request merged**